### ISSUE TITLE
"Unexpected token ." when using optional chaining

### ISSUE THREAD

**Issue by HopperElec** [sveltejs/eslint-plugin-svelte#611]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I'm using [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte). (`*.svelte` file linting does not work with the parser alone. You should also use eslint-plugin-svelte with it.)
- [X] I'm sure the problem is a parser problem. (If you are not sure, search for the issue in [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte) repo and open the issue in [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte) repo if there is no solution.
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.

### What version of ESLint are you using?

8.53.0

### What version of `eslint-plugin-svelte` and ` svelte-eslint-parser` are you using?

- svelte-eslint-parser@0.33.1
- eslint-plugin-svelte@2.53.0

### What did you do?

package.json
```json
{
  "scripts": {
    "lint": "npx eslint ."
  },
  "devDependencies": {
    "eslint": "^8.53.0",
    "eslint-plugin-svelte": "^2.35.0",
    "svelte-eslint-parser": "^0.33.1"
  }
}
```
---
.eslintrc.js
```js
module.exports = {
    extends: [
        'plugin:svelte/recommended'
    ],
};
```
---
test.svelte
```svelte
<script>
  const exampleObject = {
    exampleOptionaKey: "example optional value"
  };
  exampleObject?.exampleOptionaKey;
</script>
```

### What did you expect to happen?

No error, because that is valid JS syntax

### What actually happened?

...\src\test.svelte                                                                                                                                                                                  
  5:19  error  Parsing error: Unexpected token .

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/HopperElec/svelte-eslint-parser-mre

**Comment by JounQin** [comment]
You need to set a correct `ecmaVersion` for ESLint. Maybe we can add `ecmaVersion: 'latest'` into the base config? cc @ota-meshi

**Comment by ota-meshi** [comment]
This plugin still supports ESLint v7, so we can't use `"latest"`. The next major version will drop support for old ESLint and we'll be able to use it.

**Comment by HopperElec** [comment]
Ah, thank you! For some reason I thought that parser-related settings in the ESLint root only applied to the the built-in parser, hence me thinking it was an issue with this plugin. I think if I had a bit more experience with ESLint I probably would have recognized this, so I don't think this really needs "fixed" by the plugin.
